### PR TITLE
Customise the service name and service namespace

### PR DESCRIPTION
**Pull request by v1v**
- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your master branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [x] Link to relevant issues in GitHub or Jira
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [x] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### What

Enable the customisation for the service name and service namespace in JCasC and through the UI global setting.

It keeps the previous default behaviour of using `jenkins`.

### Why

Support CI at scale that identify the service for each Jenkins instance if needed.

### Issues

Caused by https://github.com/jenkinsci/opentelemetry-plugin/issues/88

### Tests

Manually tested 

Kapture 2021-05-06 at 17 17 12

And how does it look like in the APM UI if I configured the changes with JCasC?

```yaml
  openTelemetry:
    authentication:
      bearerTokenAuthentication:
        tokenId: "apm-server-token"
    endpoint: "https://***"
    exporterIntervalMillis: 60000
    exporterTimeoutMillis: 30000
    ignoredSteps: "dir,echo,isUnix,pwd,properties"
    observabilityBackends:
    - elastic:
        kibanaBaseUrl: "https://***"
        name: "Elastic Observability"
    serviceName: "v1v-my-jenkins"
    serviceNamespace: "v1v-jenkins"
```

A new service was added:

image

and the namespace too:

image

### CODE REVIEW

File: src/main/java/io/jenkins/plugins/opentelemetry/JenkinsOpenTelemetryPluginConfiguration.java [hunk 1]
```diff
@@ -253,6 +258,34 @@ public StepPlugin findStepPluginOrDefault(@Nonnull String stepName, @Nullable St
         return data;
     }
 
+    /**
+     * @see io.opentelemetry.semconv.resource.attributes.ResourceAttributes#SERVICE_NAME
+     */
+    @CheckForNull
```
Comment (cyrille-leclerc) [inline src/main/java/io/jenkins/plugins/opentelemetry/JenkinsOpenTelemetryPluginConfiguration.java:264]: the code below can't return `null`

File: src/main/java/io/jenkins/plugins/opentelemetry/JenkinsOpenTelemetryPluginConfiguration.java [hunk 2]
```diff
@@ -253,6 +258,34 @@ public StepPlugin findStepPluginOrDefault(@Nonnull String stepName, @Nullable St
...
+    public String getServiceName() {
+        return (Strings.isNullOrEmpty(this.serviceName)) ? JenkinsOtelSemanticAttributes.JENKINS : this.serviceName;
+    }
+
+    @DataBoundSetter
+    public void setServiceName(String serviceName) {
```
Comment (cyrille-leclerc) [inline src/main/java/io/jenkins/plugins/opentelemetry/JenkinsOpenTelemetryPluginConfiguration.java:270]: Is it `@Nullable` or `@NonNull`?

File: src/main/java/io/jenkins/plugins/opentelemetry/JenkinsOpenTelemetryPluginConfiguration.java [hunk 3]
```diff
@@ -253,6 +258,34 @@ public StepPlugin findStepPluginOrDefault(@Nonnull String stepName, @Nullable St
...
+        this.serviceName = serviceName;
+        initializeOpenTelemetry();
+    }
+
+    /**
+     * @see io.opentelemetry.semconv.resource.attributes.ResourceAttributes#SERVICE_NAMESPACE
+     */
+    @CheckForNull
+    public String getServiceNamespace() {
```
Comment (cyrille-leclerc) [inline src/main/java/io/jenkins/plugins/opentelemetry/JenkinsOpenTelemetryPluginConfiguration.java:279]: Ditto, returned value can't be `null`

File: src/main/java/io/jenkins/plugins/opentelemetry/JenkinsOpenTelemetryPluginConfiguration.java [hunk 4]
```diff
@@ -253,6 +258,34 @@ public StepPlugin findStepPluginOrDefault(@Nonnull String stepName, @Nullable St
...
+        return (Strings.isNullOrEmpty(this.serviceNamespace)) ? JenkinsOtelSemanticAttributes.JENKINS : this.serviceNamespace;
+    }
+
+    @DataBoundSetter
+    public void setServiceNamespace(String serviceNamespace) {
```
Comment (cyrille-leclerc) [inline src/main/java/io/jenkins/plugins/opentelemetry/JenkinsOpenTelemetryPluginConfiguration.java:284]: Ditto: is it `@Nullable` or `@NonNull`?

**Review by cyrille-leclerc — COMMENTED** [review]
Looks great!

File: src/main/java/io/jenkins/plugins/opentelemetry/JenkinsOpenTelemetryPluginConfiguration.java (hunk 2 above)
Comment (v1v) [reply src/main/java/io/jenkins/plugins/opentelemetry/JenkinsOpenTelemetryPluginConfiguration.java:270]: `Nullable` thought the getter will return `jenkins` if so

Comment (v1v) [reply src/main/java/io/jenkins/plugins/opentelemetry/JenkinsOpenTelemetryPluginConfiguration.java:269]: Do you mean to add the @Nullable annotation in the signature?

```suggestion
    public void setServiceName(@Nullable String serviceName) {
```

File: .editorconfig [hunk 5]
```diff
@@ -0,0 +1,8 @@
+[*]
```
Comment (v1v) [inline .editorconfig:1]: I was getting crazy with some weird tab ident in my IDE, so I've decided to add this

**Pull request merged**